### DISCUSSION TITLE
Setting Up

### DISCUSSION THREAD

**Discussion by Ro-yeee** [SebastianJames55/mind-reader#30] [General]
Hey @sebastian-james , in the steps that you have mentioned in readme, you have asked to create a file named "venv_vars.bat" inside "\mind-reader-venv\Scripts\" , but Scripts Folder does not exist , so do i have to create that ?. Also do i have to create "activate.bat" inside "\mind-reader-venv\Scripts\" ?

**Comment by SebastianJames55** [comment]
Hi @Ro-yeee, the readme is specific to Windows. Please check how to create [virtual environment](https://www.geeksforgeeks.org/python-virtual-environment/) depending on your OS. 
Thanks for pointing this out. I'm sorry for the miss, I'll update the readme. 
Please let me know if there's any other points to be addressed as well.

**Reply by Ro-yeee** [reply]
So i have setup the mindsdb server (there is some change in creating the virtual environment for mac) . But mind-reader is not starting up and its showing there is no module named openai.

**Reply by SebastianJames55** [reply]
The mindsdb server has the openai integration and mind-reader is using this. 
Within mindsdb venv you have to install the dependencies at `mindsdb/integrations/handlers/openai_handler/requirements.txt` as mentioned in the [readme](https://github.com/SebastianJames55/mind-reader/blob/dev/README.md).

**Reply by Ro-yeee** [reply]
How to install those dependencies?

**Reply by SebastianJames55** [reply]
Make sure you are in the mindsdb venv, & navigate to `mindsdb/integrations/handlers/openai_handler` then run `pip install requirements.txt`

**Reply by Ro-yeee** [reply]
Ya but  it shows this path "mindsdb/integrations/handlers/openai_handler" does not exist . There is no folder named Integrations inside of mindsdb , i dont know whether its a mac specific thing.

**Reply by SebastianJames55** [reply]
Can you ensure you did the [mindsdb setup from source](https://docs.mindsdb.com/setup/self-hosted/pip/source) as per the readme & not as [pip on MacOS](https://docs.mindsdb.com/setup/self-hosted/pip/macos)? 

As a side note, this sparked an idea - to include mindsdb as a dependency in mind-reader requirements. I'll try that & see if it's possible.

**Reply by SebastianJames55** [reply]
> Make sure you are in the mindsdb venv, & navigate to `mindsdb/integrations/handlers/openai_handler` then run `pip install requirements.txt`

`Make sure you are in the mindsdb venv` -> I meant you should have activated the venv for mindsdb & mindsdb is using that venv. 
Make sure you are in the root of the mindsdb project folder while searching for `mindsdb/integrations/handlers/openai_handler` directory. 
The venv will have auto generated files.
